README.md

```diff
@@ -215,7 +215,7 @@ Some platforms have specific requirements:
 For building, it requires:
 
 * Gradle 5.6
-* Kotlin 1.5+ (1.6+ for native)
+* Kotlin 1.6+ (1.7+ for native)
```

Comment:
Actually, 1.5+ works for building, but needs `-Xskip-metadata-version-check` compiler flag, otherwise it fails with errors like `org.jetbrains.kotlin/kotlin-stdlib-common/1.7.10/bac80c520d0a9e3f3673bc2658c6ed02ef45a76a/kotlin-stdlib-common-1.7.10.jar!/META-INF/kotlin-stdlib-common.kotlin_module: Module was compiled with an incompatible version of Kotlin. The binary version of its metadata is 1.7.1, expected version is 1.5.1.`

Reply:
My understanding is that this ensures Apollo Kotlin doesn't use any >1.5 language feature, but the error above is about trying to use the Kotlin stdlib 1.7 (a dependency of Apollo Kotlin), on a project using 1.5. Now I'm not sure if removing the warning with the flag is really safe. It worked on my little sample project 😅

Instead of this flag, another way is to pin the stdlib to 1.5 - and Apollo Kotlin will work thanks to being built with `-api-version 1.5`. But that's probably dangerous because of other dependencies that may not use this flag...

Reply:
> the error above is about trying to use the Kotlin stdlib 1.7 (a dependency of Apollo Kotlin), on a project using 1.5.

Aah yes of course! Thanks for the reminder. Damn this stuff is h ard!

> Instead of this flag, another way is to pin the stdlib to 1.5 - and Apollo Kotlin will work thanks to being built with -api-version 1.5. But that's probably dangerous because of other dependencies that may not use this flag...

No need to pin, just set

```
kotlin.stdlib.default.dependency=false
```

And we will use the "biggest" kotlin-stdlib version. `apollo-compiler` should use 1.7 because of KotlinPoet. `apollo-gradle-plugin` should use nothing because it's relocated. Everything else will keep using 1.6 until we start using a dependency that requires 1.7.

Reply:
PS: I would expect KGP to add the stdlib that's matching the `apiVersion` by default actually 🤔 . I wonder if this should be considered a bug.

Reply:
> kotlin.stdlib.default.dependency=false

Looks like it's working! I had trouble with JS but explicitly adding the `stdlib-js` dependency fixed it (I think it shouldn't be necessary, so it looks like a bug). Using `1.6.21` (most recent version compatible with 1.5).

> I would expect KGP to add the stdlib that's matching the apiVersion by default actually 🤔 . I wonder if this should be considered a bug.

Agreed ✅ Filed as [KT-53462](https://youtrack.jetbrains.com/issue/KT-53462).